gnovm/cmd/gno/clean.go

```diff
@@ -0,0 +1,116 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"flag"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/gnolang/gno/gnovm/pkg/gnomod"
+	"github.com/gnolang/gno/tm2/pkg/commands"
+	"github.com/gnolang/gno/tm2/pkg/crypto/keys/client"
+)
+
+type cleanCfg struct {
+	dryRun   bool // clean -n flag
+	verbose  bool // clean -x flag
+	modCache bool // clean -modcache flag
+}
+
+func newCleanCmd(io *commands.IO) *commands.Command {
+	cfg := &cleanCfg{}
+
+	return commands.NewCommand(
+		commands.Metadata{
+			Name:       "clean",
+			ShortUsage: "clean [flags]",
+			ShortHelp:  "Removes generated files and cached data",
+		},
+		cfg,
+		func(ctx context.Context, args []string) error {
+			return execClean(cfg, args, io)
+		},
+	)
+}
+
+func (c *cleanCfg) RegisterFlags(fs *flag.FlagSet) {
+	fs.BoolVar(
+		&c.dryRun,
+		"n",
+		false,
+		"print remove commands it would execute, but not run them",
+	)
+
+	fs.BoolVar(
+		&c.verbose,
+		"x",
+		false,
+		"print remove commands as it executes them",
+	)
+
+	fs.BoolVar(
+		&c.modCache,
+		"modcache",
+		false,
+		"remove the entire module download cache",
+	)
+}
+
+func execClean(cfg *cleanCfg, args []string, io *commands.IO) error {
+	if len(args) > 0 {
+		return flag.ErrHelp
+	}
+
+	path, err := os.Getwd()
+	if err != nil {
+		return err
+	}
+	modDir, err := gnomod.FindRootDir(path)
+	if err != nil {
+		return errors.New("not a gno module")
+	}
```

Comment:
swallowed error

Reply:
Did it intentionally. Not confident about it anymore. 
Maybe?
```go
if err != nil {
	return errors.New("not a gno module; %w", err)
}
```

Reply:
yeah, looks better. These swallowed errors might cause many problems in the future from my experience.

Reply:
Should be resolved by: 459fab1